src/typing/constraint_js.ml

```diff
@@ -507,7 +507,16 @@ type block_entry = {
   def_loc: Spider_monkey_ast.Loc.t option;
   for_type: bool;
 }
-type block = block_entry SMap.t ref
+
+type block_scope =
```

Comment:
Let's call this "scope_kind" (i.e. odd that the name seems to imply one of the variants but not the other -- in fact the issue really is that "block" means two different things now, but more on that in the next comment...)